File: 3639-zero-array-transformation-i/3639-zero-array-transformation-i.py
class Solution:
    def isZeroArray(self, nums: List[int], queries: List[List[int]]) -> bool:
        n = len(nums)
        res = [0] * (n + 1)
        for q in queries:
            l = q[0]
            r = q[1]
            res[l] += 1
            res[r + 1] -= 1
        
        for i in range(1, n + 1):
            res[i] += res[i - 1]
        
        for i in range(n):
            if res[i] < nums[i]:
                return False
        
        return True

File: 3639-zero-array-transformation-i/3639-zero-array-transformation-i.go
func isZeroArray(nums []int, queries [][]int) bool {
    n := len(nums)
    del := make([]int, n + 1)
    
    for _, q := range queries {
        l := q[0]
        r := q[1]
        del[l]++
        del[r + 1]--
    }
    
    for i := 1; i <= n; i++ {
        del[i] += del[i - 1]
    }
    
    for i := 0; i < n; i++ {
        if del[i] < nums[i] {
            return false
        }
    }
    return true
}